apps/veil/src/pages/trade/ui/order-form/store/LimitOrderFormStore.ts

```diff
@@ -73,23 +73,29 @@ export class LimitOrderFormStore {
   }
 
   get baseInput(): string {
-    return this._input.inputA;
+    return this._input.inputA.length
+      ? round({ value: Number(this._input.inputA), decimals: this._baseAsset?.exponent ?? 6 })
+      : '';
   }
 
   setBaseInput = (x: string) => {
     this._input.inputA = x;
   };
 
   get quoteInput(): string {
-    return this._input.inputB;
+    return this._input.inputB.length
+      ? round({ value: Number(this._input.inputB), decimals: this._quoteAsset?.exponent ?? 6 })
+      : '';
   }
 
   setQuoteInput = (x: string) => {
     this._input.inputB = x;
   };
 
   get priceInput(): string {
-    return this._priceInput;
+    return this._priceInput.length
+      ? round({ value: Number(this._priceInput), decimals: this._quoteAsset?.exponent ?? 6 })
+      : '';
```

Comment:
Same here ; we want to move presentation to the components displaying this information.